src/ocean_emulators/datasets.py

```diff
@@ -266,11 +271,15 @@ def __getitem__(self, idx):
 
     def _get_x_index(self, idx):
         if isinstance(idx, slice):
-            if idx.start < 0 or idx.stop < 0 or idx.step < 0:
+            if (
+                (idx.start and idx.start < 0)
+                or (idx.stop and idx.stop < 0)
+                or (idx.step and idx.step < 0)
+            ):
                 raise IndexError("Sorry, negative indexing is not supported!")
-            elif idx.start >= self.size or idx.stop >= self.size:
-                raise IndexError(f"Index {idx} out of range with size {self.size}")
-            elif idx.start is None and idx.stop is None:
+            if idx.step is None:
+                idx = slice(idx.start, idx.stop, 1)
+            if idx.start is None and idx.stop is None:
```

Comment:
I think we could go even further:
```suggestion
            if idx.start is None:
                idx = slice(0, idx.stop, idx.step)
            if idx.stop is None:
                idx = slice(idx.start, self.size, idx.step)
```

Reply:
(probably not a good idea to accept this suggestion; I couldn't modify the code below)

Reply:
I think it is good the way it is right now, There won't be any cases missed (Unless I am mistaken?)